addons/project/views/project_task_views.xml

```diff
@@ -19,6 +19,7 @@
                     <filter string="Favorite Projects" name="favorite_projects" domain="[('project_id.is_favorite', '=', True)]" invisible="context.get('default_project_id')"/>
                     <separator/>
                     <filter string="Starred Tasks" name="starred_tasks" domain="[('priority', '=', '1')]"/>
+                    <filter string="Private Tasks" name="private_tasks" domain="[('project_id', '=', False)]" invisible="context.get('default_project_id')"/>
```

Comment:
Maybe don't display this filter in the reporting views, since we only display non private tasks there?
In that case, the filter should be added in `view_task_search_form` and `project_task_view_search_fsm`